OgreMain/include/Ogre.i

```diff
@@ -497,6 +509,22 @@ SHARED_PTR(Material);
 %include "OgreMaterialManager.h"
 %include "OgreRenderable.h"
 %include "OgreShadowCaster.h"
+#ifdef SWIGCSHARP
+%typemap(cscode) Ogre::MovableObject %{
+  public bool tryCastEntity(out Entity entity)
+  {
+	entity = null;
+  	if (this.getMovableType() != "Entity")
```

Comment:
assuming the check on C++ side is working, this is not needed. Also you should either go with the return value or with exception for error handling.